lib/client.js

```diff
@@ -143,6 +155,30 @@ extend(Raven.prototype, {
     return this;
   },
 
+  makeErrorHandler: function () {
+    var self = this;
+    var calledUncaughtHandler = false;
+    var calledFatalError = false;
+    var firstError;
+    return function (err) {
+      if (!calledUncaughtHandler) {
+        firstError = err;
+        calledUncaughtHandler = true;
+        self.captureException(err, function (sendErr, eventId) {
+          calledFatalError = true;
+          self.onFatalError(err, sendErr, eventId);
+        });
+      } else if (!calledFatalError) {
+        self.onFatalError(firstError, err);
+      } else {
+        // we hit an error *after* calling onFatalError - pretty fucked
```

Comment:
Fair point. I ran the numbers on the codebase's current 🚽-mouth stature:

| Word | # |
| ---- | - |
| Crap | 6 |
| Fuck | 3 |
| Shit | 1 |

And for commit messages:
| Word | # |
| ---- | - |
| Crap | 0 |
| Fuck | 1 |
| Shit | 1 |

We can put things through the 🛁  in a later commit.